src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -48,6 +48,12 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
         // create the snapshot metadata object in postgres and lock it
         addStep(new CreateSnapshotMetadataStep(snapshotDao, snapshotService, snapshotReq));
 
+        // Create the IAM resource and readers for the snapshot
+        // The IAM code contains retries, so we don't make a retry rule here.
+        AuthenticatedUserRequest userReq = inputParameters.get(
+            JobMapKeys.AUTH_USER_INFO.getKeyName(), AuthenticatedUserRequest.class);
+        addStep(new SnapshotAuthzIamStep(iamClient, snapshotService, snapshotReq, userReq));
+
```

Comment:
I see no problem with this---my preference would be to add it after the BQ steps though.
Adding this above the bq validation step feels like it could slow us down---since I assume the SAM step is likely to take more time and snapshots are most likely to fail during validation or creation.